components/ConvertSteps/ManageStep.js

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useCallback, useState } from 'react'
+import PropTypes from 'prop-types'
+import Step from './Step'
+
+function ManageStep({
+  title,
+  number,
+  handleTx,
+  active,
+  onSuccess,
+  onError,
+  onHashCreation,
+  retry,
+}) {
+  const [stepStatus, setStepStatus] = useState('waiting')
+  const [hasBeenActivated, setHasBeenActivated] = useState(false)
+  const [retrying, setRetrying] = useState(false)
+  const [hash, setHash] = useState()
+
+  const handleStepSigning = useCallback(async () => {
+    setHasBeenActivated(true)
+
+    try {
+      setStepStatus('waiting')
+
+      // Awaiting confirmation
+      const transaction = await handleTx()
+
+      onHashCreation && onHashCreation(transaction.hash)
+      setHash(transaction.hash)
+
+      // Mining transaction
+      setStepStatus('working')
+      await transaction.wait()
+
+      // Success
+      setStepStatus('success')
+
+      onSuccess && onSuccess()
+    } catch (err) {
+      setStepStatus('error')
+      onError && onError()
+      console.log(err)
```

Comment:
```suggestion
      console.error(err)
```

We may also want to provide a bit more context to the error :).